### PR TITLE
FSE: Fix template part update button render loop

### LINKED ISSUE #34705: Full Site Editing: TemplateUpdateConfirmationButton causes rendering loop
#### Steps to reproduce
1. Check out branch follow instructions in https://github.com/Automattic/wp-calypso/tree/master/apps/full-site-editing#build-system. Install the FSE plugin, and activate it on local gutenberg dev
2. Add console log to /full-site-editing/full-site-editing-plugin/full-site-editing/plugins/template-update-confirmation/button.js componentDidUpdate
3. Edit a template part, eg. Header or Footer with console open - console.log will show the component constantly rerendering
4.  Comment the following in componentDidUpdate and re-render stops - so looks like the onResize method is causing the render loop somehow.

```
	if ( props.isFullScreen !== prevProps.isFullScreen ) {
			// this ensures the button is repositioned properly when toggling fullscreen mode
			setTimeout( () => this.onResize(), 1 );
		}
```

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

* Render the FSE template part update button into the editor toolbar to prevent the need to listen for changes in full page editing mode  which was causing a re-render loop

#### Testing instructions

* Check out branch follow instructions in https://github.com/Automattic/wp-calypso/tree/master/apps/full-site-editing#build-system. Install the FSE plugin, and activate it on local gutenberg dev

* Edit a footer or header template part and check that the Update button causes update warning panel to appear in both normal and full screen edit. Also check everything works correctly till in different screen sizes

Fixes #34705

### CODE REVIEW

**Comment by glendaviesnz** [comment]
It may not be worth landing this if we go with Is it worth modifying this to the notice formatting suggestion, or close it and go with #34728 as the button override won't be needed for that approach.

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/template-update-confirmation/button.js [hunk 1]
```diff
@@ -23,37 +23,29 @@ const initialState = {
...
 
-	onResize() {
+	onSetButtonStyles() {
 		const originalButton = this.getOriginalButton();
 		let { buttonStyle } = initialState;
 		if ( isNil( originalButton ) || ! ( 'getBoundingClientRect' in originalButton ) ) {
 			// if it's not there, might need a timeout to await it?
 			return this.setState( { buttonStyle } );
 		}
-		const rect = originalButton.getBoundingClientRect();
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/template-update-confirmation/button.js]: Nice, this call does trigger a reflow, and so tend to be expensive to call.

**Comment by gwwar** [comment]
Quickly tested this, and it looks like the publish warning doesn't render appropriately if settings are open.

> It may not be worth landing this 

If we don't think the panel override is going to stick around, I think it's fine to close out. We may want to revisit if not otherwise.

**Comment by glendaviesnz** [comment]
Going to close this in favour of #34728

**Pull request closed without merging**